Xcode 14.1 C++ build failures

```
src/IceSSL/SecureTransportUtil.cpp:215:19: error: 'SecKeychainCopyDefault' is deprecated: first deprecated in macOS 10.10 - SecKeychain is deprecated [-Werror,-Wdeprecated-declarations]
        if((err = SecKeychainCopyDefault(&keychain.get())))
                  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Security.framework/Headers/SecKeychain.h:384:10: note: 'SecKeychainCopyDefault' has been explicitly marked deprecated here
OSStatus SecKeychainCopyDefault(SecKeychainRef * __nonnull CF_RETURNS_RETAINED keychain)
         ^
src/IceSSL/SecureTransportUtil.cpp:235:19: error: 'SecKeychainOpen' is deprecated: first deprecated in macOS 10.10 - SecKeychain is deprecated [-Werror,-Wdeprecated-declarations]
        if((err = SecKeychainOpen(keychainPath.c_str(), &keychain.get())))
                  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Security.framework/Headers/SecKeychain.h:294:10: note: 'SecKeychainOpen' has been explicitly marked deprecated here
OSStatus SecKeychainOpen(const char *pathName, SecKeychainRef * __nonnull CF_RETURNS_RETAINED keychain)
         ^
src/IceSSL/SecureTransportUtil.cpp:243:11: error: 'SecKeychainGetStatus' is deprecated: first deprecated in macOS 10.10 - SecKeychain is deprecated [-Werror,-Wdeprecated-declarations]
    err = SecKeychainGetStatus(keychain.get(), &status);
          ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Security.framework/Headers/SecKeychain.h:462:10: note: 'SecKeychainGetStatus' has been explicitly marked deprecated here
OSStatus SecKeychainGetStatus(SecKeychainRef __nullable keychain, SecKeychainStatus *keychainStatus)
         ^
src/IceSSL/SecureTransportUtil.cpp:247:19: error: 'SecKeychainUnlock' is deprecated: first deprecated in macOS 10.10 - SecKeychain is deprecated [-Werror,-Wdeprecated-declarations]
        if((err = SecKeychainUnlock(keychain.get(), static_cast<UInt32>(keychainPassword.size()), pass, pass != 0)))
                  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Security.framework/Headers/SecKeychain.h:355:10: note: 'SecKeychainUnlock' has been explicitly marked deprecated here
OSStatus SecKeychainUnlock(SecKeychainRef __nullable keychain, UInt32 passwordLength, const void * __nullable password, Boolean usePassword)
         ^
src/IceSSL/SecureTransportUtil.cpp:257:19: error: 'SecKeychainCreate' is deprecated: first deprecated in macOS 10.10 - SecKeychain is deprecated [-Werror,-Wdeprecated-declarations]
        if((err = SecKeychainCreate(keychainPath.c_str(),
                  ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Security.framework/Headers/SecKeychain.h:309:10: note: 'SecKeychainCreate' has been explicitly marked deprecated here
OSStatus SecKeychainCreate(const char *pathName, UInt32 passwordLength, const void * __nullable password, Boolean promptUser, SecAccessRef __nullable initialAccess, SecKeychainRef * __nonnull CF_RETURNS_RETAINED keychain)
         ^
src/IceSSL/SecureTransportUtil.cpp:278:15: error: 'SecKeychainSetSettings' is deprecated: first deprecated in macOS 10.10 - SecKeychain is deprecated [-Werror,-Wdeprecated-declarations]
    if((err = SecKeychainSetSettings(keychain.get(), &settings)))
              ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Security.framework/Headers/SecKeychain.h:330:10: note: 'SecKeychainSetSettings' has been explicitly marked deprecated here
OSStatus SecKeychainSetSettings(SecKeychainRef __nullable keychain, const SecKeychainSettings *newSettings)
         ^
src/IceSSL/SecureTransportUtil.cpp:433:5: error: 'SecKeychainItemModifyAttributesAndData' is deprecated: first deprecated in macOS 10.10 - SecKeychain is deprecated [-Werror,-Wdeprecated-declarations]
    SecKeychainItemModifyAttributesAndData(reinterpret_cast<SecKeychainItemRef>(key.get()), &attrs, 0, 0);
    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/Security.framework/Headers/SecKeychainItem.h:165:10: note: 'SecKeychainItemModifyAttributesAndData' has been explicitly marked deprecated here
OSStatus SecKeychainItemModifyAttributesAndData(SecKeychainItemRef itemRef, const SecKeychainAttributeList * __nullable attrList, UInt32 length, const void * __nullable data)
         ^
7 errors generated.
```

--- a/cpp/src/IceSSL/SecureTransportUtil.cpp
+++ b/cpp/src/IceSSL/SecureTransportUtil.cpp
@@ -23,6 +23,10 @@ using namespace IceSSL;
 using namespace IceSSL::SecureTransport;
 using namespace std;
 
+#if defined(__clang__)
+#   pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#endif
+
 namespace
 {
 

